dips/dip-161.md

```diff
@@ -0,0 +1,51 @@
+---
+dip: 161
+title: Optional Result Field in Off-Chain Protocol
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 04/08/2020
+last_updated: 04/08/2021
+issue: https://github.com/diem/dip/issues/161
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Overview
+
+This DIP describes an optional result field in Off-Chain command response, and is an extension of [DIP-1](https://github.com/diem/dip/blob/main/dips/dip-1.mdx).
+Both the Off-Chain Protocol and Travel Rule Exchange (or PaymentCommand) are defined in DIP-1. The Off-Chain Protocol in its current definition lacks a way to get back a result that's more informative than a success or failure status. Many types of communication benefit from having richer information in responses that may prevent the need for another message exchange, leading to simpler protocol and  reduced latency. 
+
+In order to support this, the `CommandResponseObject` can leverage an optional `result` field. The `result` field, if defined, must contain `_ObjectType` that uniquely defines other fields within the result.
+
+Here's an example from DIP-10:
+```
+{
+   "_ObjectType": "CommandResponseObject",
+    "status": "success",
+    "result": {
+	    "_ObjectType": "ReferenceIDCommandResponse",
+	    "receiver_address": "c5ab123458df0003415689adbb47326d",
+    },
+    "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+}
+```
+
+**CommandResponseObject:**
+
+| Field 	    | Type 	     | Required? 	| Description 	           |
+|-------	    |------	     |-----------	|-------------	           |
+| _ObjectType   | str        | Y | Fixed value: `CommandResponseObject`|
+| status       | str | Y | Either `success` or `failure`. |
+| result       | CommandResultObject() | N | The Result object of response. |
+| cid           | str         | Y            | A unique identifier for the Command. Should be a UUID according to [RFC4122](https://tools.ietf.org/html/rfc4122) with "-"'s included. |
```

Comment:
cid is optional defined in dip-1

Reply:
Yup, look at what I did in the spec: https://github.com/diem/diem/tree/main/specifications/transactions/offchain-api